crypto/statetrie/nibbles/nibbles_test.go

```diff
@@ -0,0 +1,189 @@
+// Copyright (C) 2019-2023 Algorand, Inc.
+// This file is part of go-algorand
+//
+// go-algorand is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// go-algorand is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with go-algorand.  If not, see <https://www.gnu.org/licenses/>.
+
+package nibbles
+
+import (
+	"bytes"
+	"github.com/algorand/go-algorand/test/partitiontest"
+	"github.com/stretchr/testify/require"
+	"math/rand"
+	"testing"
+	"time"
+)
+
+func TestNibbles(t *testing.T) { // nolint:paralleltest // Serial tests for trie for the moment
+	partitiontest.PartitionTest(t)
+	t.Parallel()
+
+	rand.Seed(time.Now().UnixNano())
```

Comment:
```suggestion
	seed := time.Now().UnixNano()
	rand.Seed(seed)
	defer func() {
		if t.Failed() {
			t.Logf("Seed used: %d\n", seed)
		}
	}
```

Reply:
thanks [#ff75a3e](https://github.com/algorand/go-algorand/pull/5719/commits/ff75a3e4751c3b9ca5778171447f02b849f2a642), I tried to break it out to its own test, and it creates a local rand now.